packages/@aws-cdk/aws-ecr-assets/lib/image-asset.ts

```diff
@@ -97,20 +96,30 @@ export class DockerImageAsset extends CoreConstruct implements assets.IAsset {
       throw new Error(`Cannot find file at ${file}`);
     }
 
+    let ignoreMode = props.ignoreMode || IgnoreMode.DOCKER;
+
+    if (ignoreMode != IgnoreMode.DOCKER) {
```

Comment:
Why this restriction? I am okay with the default being `DOCKER`, but can't we support the old ones as well in case people depend on it?

Reply:
The reason I originally did this is because it feels plain wrong to process `.dockerignore` patterns with anything other than `IgnoreMode.DOCKER`.

This is the core thing that relates to the questions of backward compatibility, meaning, I'm not super confident how existing _user-supplied_ `exclude` patterns will behave when flipped from the current `IgnoreMode.GLOB` (`minimatch`) to `IgnoreMode.DOCKER`, because both have different nuanced behavior.

So with this in mind, in the event that users may have become inadvertently dependent on the current broken behavior, maybe it _is_ a good idea to still allow that old behavior as an option. Must we go even further and leave legacy as the default, with `IgnoreMode.DOCKER` as opt-in? It would be safest, though it would leave major out-of-the-box ergonomics on the table.

See 2f0b88f where I removed the restriction.

Another option is to default `ignoreMode` to the legacy `IgnoreMode.GLOB` if `exclude` was passed by the user but `ignoreMode` wasn't, since it likely means that they may be dependent on the legacy behavior.

Then again, maybe just letting users opt back into the legacy behavior if they run into issues is good enough. What do you think?

Reply:
I went a little more into this in the comment below.

https://github.com/aws/aws-cdk/pull/10922#issuecomment-717875238